### DISCUSSION TITLE
Usefulness of a text-to-speech feature?

### DISCUSSION THREAD

**Discussion by seongryoo** [vkbo/novelWriter#1966] [Ideas]
When I'm writing, I like to use text-to-speech software to read aloud my drafts. I find it helpful when editing, since hearing what I've written can often reveal typos and mistakes which my eyes just skipped over. It's also just like a nice 'sanity check.'

Currently, my text-to-speech workflow is to export my novelWriter project as a .pdf and then open said .pdf on Microsoft Edge to take advantage of the browser's native ReadAloud feature, which has surprisingly good voices. Would there be any interest in adding a minimal text-to-speech feature to novelWriter? I think it would be amazing if you could highlight a portion of text and have it be read aloud, even by a really basic robotic voice.

**Comment by peter88213** [comment]
You might want to take a look at the LibreOffice "Read Text" extension. This works pretty well for me. 
https://extensions.libreoffice.org/en/extensions/show/read-text
I can imagine that this will simplify your workflow a little. 

If something like this should be implemented in *novelWriter*, a glance at the extension's Python code might be helpful:
https://github.com/jimholgate/readtextextension

Edit:
It's also quite possible that such a feature comes with Qt.

**Reply by vkbo** [reply]
Qt does indeed have a wrapper to call the OS' text to speech engine, whatever that may be. It has a [QTextToSpeech](https://doc.qt.io/qt-5/qtexttospeech.html) class.

@seongryoo, feel free to make a feature request for it. It looks like it may be reasonably straightforward to add, based on the description in the link.

**Reply by seongryoo** [reply]
Got it, thank you both for the insights!
